### ISSUE TITLE
[FEATURE] Add `no-duplicate-in-head` rule

### ISSUE THREAD

**Issue by yeonjuan** [yeonjuan/html-eslint#328]
**Is your feature request related to a problem? Please describe.**

The no-duplicate-meta rule would ensure that `<meta>` elements within the `<head>` section do not duplicate the same `name` or `property`

**Additional context**

Incorrect example
```html
<head>
  <meta name="description" content="First description">
  <meta name="description" content="Second description">
</head>
```

correct example
```html
<head>
  <meta name="description" content="Only one description">
</head>
```

**Comment by tausiq2003** [comment]
hey, I am now working on this issue, so you mentioned `<meta>` elements should not contain same `name` or `property`, do you mean the `property` attribute or properties in general, my guess here is `property` attribute here. Please reply kindly.

**Comment by yeonjuan** [comment]
Hi @tausiq2003 Thank you for sharing this while you're working on it.

At first, I came up with a simple rule that reports tags under the `<head>` element if they have duplicate values in either the `name` or `property` attributes.

```html
<head>
    <meta name="viewport" content=""> 
    <meta name="viewport" content=""> 
</head>
```

However, on second thought, there are some tags that shouldn't be duplicated under the `<head>` element even if they don't use name or property—such as `<title>`, `<base>`, or `<link rel="stylesheet">` tags with the same href, and so on. Ultimately, it would be nice to handle these cases as well, but it's not necessary to cover everything in the initial implementation.

I think `no-duplicate-in-head` would be a good name for the rule.

**Comment by Sendi0011** [comment]
I will like to work on this. 
ETA: 12 hours

**Comment by Sendi0011** [comment]
Thank you sir. I will let you know if I have any questions

**Comment by yeonjuan** [comment]
fixed: https://github.com/yeonjuan/html-eslint/pull/362
